code/GroundStation.ino

```diff
@@ -0,0 +1,293 @@
+      inNum++;
+      break;   
+    }
+    inTemp += in[inNum];
+    inNum++;
+    if(inNum > in.length()){
+      Serial.print(timeError);
+      return;
+    }
+  }
+  theTime.inMinute = inTemp.toInt();
+  inTemp = "";
+  
+  while(true){
+    if(in[inNum] == ';'){
+      inNum++;
+      break;   
+    }
+    inTemp += in[inNum];
+    inNum++;
+    if(inNum > in.length()){
+      Serial.print(timeError);
+      break;
+    }
+  }
+  theTime.inSecond = inTemp.toInt();
+  inTemp = "";
+  
+  while(true){
+    if(inNum >= in.length()-1){
+      inNum++;
+      break;   
+    }
+    inTemp += in[inNum];
+    inNum++;
+  }
+  theTime.inMilisecond = inTemp.toInt();
+
+  theTime.hour = theTime.inHour;
+  theTime.minute = theTime.inMinute;
+  theTime.second = theTime.inSecond;
+  theTime.milisecond = theTime.inMilisecond;
+}
+
+int fileNumberS = 0;
+int fileNumberR = 0;
+
+void saveData(String inData, byte type){
+  String fileName = "";
+  if(type == 0){
+    fileName = "S";
+    fileName += fileNumberS;
+  }else{
+    fileName = "R";
+    fileName += fileNumberR;
+  }
+  fileName += ".txt";
+  usedFile = SD.open(fileName, FILE_WRITE);
+  if(usedFile){
+    if(!usedFile.print(inData)){
+      Serial.println(sdError);
+    }
+  }
+  
+  usedFile.close();
+}
+
+String showTime(){
+  String out = "Time: ";
+  out += theTime.hour;
+  out += ":";
+  out += theTime.minute;
+  out += ":";
+  out += theTime.second;
+  out += ";";
+  out += theTime.milisecond;
+  return out;
+}
+
+class SensorData{
+  public:
+    float pressure;
+    float altitude;
+    float temperature;
+};
+
+SensorData sensorData;
+int saveSensorTime;
+
+String showSensor(){
+  String out = "Pressure: ";
+  out += sensorData.pressure;
+  out += ", Altitude: ";
+  out += sensorData.altitude;
+  out += ", Temperature: ";
+  out += sensorData.temperature;
+  return out;
+}
+
+void getSensorData(){
+  sensorData.pressure = sensor.readPressureMillibars();
+  sensorData.altitude = sensor.pressureToAltitudeMeters(sensorData.pressure);
+  sensorData.temperature = sensor.readTemperatureC();
+}
+
+void getSaveSensorTime(){
+  Serial.print(getSaveTime);
+  int waitForData = 0;
+  String in = autoSaveTime;
+  while(Serial.available() == 0 && waitForData < serialDelay){
+    waitForData = millis() - startTime;
+  }
+  if(Serial.available() > 0){
+    in = Serial.readString();
+  }
+  saveSensorTime = in.toInt();
+}
+
+void countFiles(){
+  int number = 1;
+  while(true){
+    String fileName = "S/";
+    fileName += number;
+    fileName += ".txt";
+    if(!SD.exists(fileName)){
+      break;
+    }
+    number++;
+  }
+  fileNumberS = number;
+  number = 1;
+  while(true){
+    String fileName = "R/";
+    fileName += number;
+    fileName += ".txt";
+    if(!SD.exists(fileName)){
+      break;
+    }
+    number++;
+  }
+  fileNumberR = number;
+}
+
+void receiveData(){
+  if (radioSerial.available())
+    Serial.write(radioSerial.read());
+}
+
+void setup(){
+  
+  Serial.begin(9600);
+/*
```

Comment:
Zamierzasz tego używać później? Jak nie to wywal wszystkie komentarze które nie są komentarzami.